src/shared/sol-util.c

```diff
@@ -312,14 +315,37 @@ sol_util_str_split(const struct sol_str_slice slice, const char *delim, size_t m
 static int
 get_progname(char *out, size_t size)
 {
-    char cwd[PATH_MAX] = { NULL }, *execfn;
+    char cwd[PATH_MAX] = { NULL }, *execfn = NULL;
 
     if (getcwd(cwd, sizeof(cwd)) == NULL)
-        return -1;
+        return -errno;
 
+#ifdef HAVE_SYS_AUXV_H
     execfn = (char *)getauxval(AT_EXECFN);
```

Comment:
@ceolin Thinking better, getauxval() returns ptrs to static memory allocated somewhere else, and I can't do that on the clone (nor have a free() afterwards version). I'll stick with the alternate behavior in place.